### PR TITLE
781-add-action-bar-in-topology-sidebar

### LINKED ISSUE #781: Add action bar in topology sidebar
i.e.all actions in the burger menu should be actions at the top of the of the sidebar.

### PR DESCRIPTION
**Pull request by mainawycliffe**
Closes #781

### CODE REVIEW

**Comment by moshloop** [comment]
- Can make the burger menu and the sidebar actions both use the same actions object so that changes to actions only need to take place in  1 place
- Can we use the normal button component?  it is not very clear that these are buttons

**Comment by moshloop** [comment]
@mainawycliffe - I don't see any changes?  This must use the https://github.com/flanksource/flanksource-ui/tree/main/src/components/Button component, If we make styling changes to that component, they must propogate to every button.

**Comment by mainawycliffe** [comment]
@moshloop Just doing some cleanup, will be pushing the changes in the next few minutes

**Comment by mainawycliffe** [comment]
@moshloop I have made the changes, what do you think?

File: src/components/TopologySidebar/TopologyActionBar.tsx [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+  onClick = () => {},
+  icon,
+  text
+}: TopologyActionLinkProps) {
+  return (
+    <div className="flex w-full flex-col items-center p-1 h-full">
+      <Button
+        icon={icon}
+        text={text}
+        className="flex w-full h-full flex-col space-y-2 justify-center items-center btn-primary"
```
Comment (moshloop) [inline src/components/TopologySidebar/TopologyActionBar.tsx:217]: We should not be changing the class on the buttons - the only thing that should every change is the color. The style should be normal button icon on the left, text on the right - rectangle, not square

File: src/components/TopologySidebar/TopologyActionBar.tsx [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+      const navigate = useNavigate();
+
+      return (
+        <ChildComponent
+          onClick={() => navigate(`/topology/${topology.id}`)}
+          icon={icon}
+          text={text}
+        />
+      );
+    }
```
Comment (moshloop) [inline src/components/TopologySidebar/TopologyActionBar.tsx:109]: Why can't this just be onClick: {() => navigate(...)}

File: src/components/TopologySidebar/TopologyActionBar.tsx [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+  label: string;
+  /**
+   *
+   * The icon to be shown in the action item, this is a react component and can
+   * be rendered directly
+   *
+   * e.g. <Icon {...props} />
+   *
+   */
+  Icon: IconType;
```
Comment (moshloop) [inline src/components/TopologySidebar/TopologyActionBar.tsx:27]:
```suggestion
  icon: IconType;
```

File: src/components/TopologySidebar/TopologyActionBar.tsx [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+import { EvidenceType } from "../../api/services/evidence";
+import { Topology } from "../../context/TopologyPageContext";
+import { AttachEvidenceDialog } from "../AttachEvidenceDialog";
+import { Button } from "../Button";
+import TopologySnapshotModal from "../TopologyCard/TopologySnapshotModal";
+
+type TopologyActionItem = {
+  /**
+   * The label to be shown in the action item
+   */
```
Comment (moshloop) [inline src/components/TopologySidebar/TopologyActionBar.tsx:17]: Don't add comments for the obvious

File: src/components/TopologySidebar/TopologyActionBar.tsx (hunk 3 above)
Comment (mainawycliffe) [reply src/components/TopologySidebar/TopologyActionBar.tsx:27]: Due to how react works, components must start with a capital letter. Hence the options are either this or alias every time we need to use them in order to get intellisense and be recognized as components.

File: src/components/TopologySidebar/TopologyActionBar.tsx (hunk 2 above)
Comment (mainawycliffe) [reply src/components/TopologySidebar/TopologyActionBar.tsx:109]: The logic to determine URL to navigate to would also need to be extracted away (properties also don't work as they don't have access to the topology object needed to determine the component details), which would affect readability and ease of use when consuming the list and also adding new action items.

**Comment by moshloop** [comment]
@mainawycliffe can we remove the panel border, and switch to white buttons like on the incident page.

**Comment by mainawycliffe** [comment]
@moshloop you mean like this

image

**Comment by moshloop** [comment]
@mainawycliffe I don't see any buttons now ?

**Comment by mainawycliffe** [comment]
@moshloop sorry, messed the rebase, fixed it now.

**Pull request merged**